examples/ipac2018_mayes/run_fieldplots_ipac2018.py

```diff
@@ -0,0 +1,190 @@
+from impactx import ImpactX, distribution, elements
+
+# theory data from eq. (1) in Mayes, sampled to text files in
+theory = pd.read_csv("Ex_Mayes.dat", header=0, sep=r"\s+")
+x_theory = theory["x/sigmax"]
+
+
+def run_and_plot(case, r, ax):
+    """Run ImpactX for a new r and plot
+
+    Parameters
+    ----------
+    case : string
+      case name
+    r : Float
+      aspect ratio = sigma_z / sigma_perp:  range 0.01 to 10
+    ax : matplotlib.axes
+      a figure axis to plot into
+    """
+    sim = ImpactX()
+
+    # set numerical parameters and IO control
+    sim.n_cell = [128, 128, 256]  # 128^3 or higher for reasonable convergence
+    # sim.n_cell = [32, 32, 64]
+    sim.particle_shape = 2  # B-spline order
+    sim.space_charge = True
+    sim.dynamic_size = True
+    sim.prob_relative = [4.0]
+
+    # space charge solver fine-tuning
+    sim.mlmg_relative_tolerance = 1.0e-8
+    sim.mlmg_max_iters = 500
+    sim.mlmg_verbosity = 2
+
+    # beam diagnostics
+    sim.diagnostics = False
+    sim.slice_step_diagnostics = False
+
+    # domain decomposition & space charge mesh
+    sim.init_grids()
+
+    # load a cold 10 MeV electron beam
+    mass_MeV = 0.510998950  # electron mass in MeV/c^2
+    energy_MeV = 1.00001 * mass_MeV  # reference energy (total)
+    bunch_charge_C = 1.0e-9  # charge in C
+    npart = int(1000000)  # number of macro particles
+
+    #   reference particle (Mayes Fig. 1 is for + charge)
+    ref = sim.particle_container().ref_particle()
+    kinetic_MeV = energy_MeV - mass_MeV
+    ref.set_charge_qe(1.0).set_mass_MeV(mass_MeV).set_kin_energy_MeV(kinetic_MeV)
+
+    #   particle bunch
+    sigma_r = 1.0e-3  # fixed at 1 mm
+    sigma_z = r * sigma_r
+    gamma = energy_MeV / mass_MeV
+    beta = (1.0 - (1.0 / gamma) ** 2) ** 0.5
+    # c0 = 2.99792458e8  # speed of light in m/s (not used)
+    sigma_t = sigma_z / (beta * gamma)  # recall t is implicitly scaled by c0
+    print(f"r={r} sigma_t={sigma_t}m")
+
+    distr = distribution.Gaussian(
+        sigmaX=sigma_r,
+        sigmaY=sigma_r,
+        sigmaT=sigma_t,
+        sigmaPx=0.0,
+        sigmaPy=0.0,
+        sigmaPt=0.0,
+    )
+    sim.add_particles(bunch_charge_C, distr, npart)
+
+    # add beam diagnostics
+    monitor = elements.BeamMonitor("monitor", backend="h5")
+
+    # design the accelerator lattice
+    sim.lattice.extend([monitor])
+
+    # run simulation
+    sim.evolve()
+
+    # plotting
+
+    # simulation data
+    F_x = sim.space_charge_field(lev=0, comp="x")  # [V/m] in the lab frame
+
+    gm = sim.Geom(lev=0)
+    dr = gm.data().CellSize()
+    # dV = np.prod(dr)  # not used
+
+    half_x, half_y, half_z = [n // 2 for n in sim.n_cell]  # order: x,y,z
+
+    # plot theory
+    p = ax.plot(x_theory, theory[case], label=case)  # E_x(sigma_r) [MV/m]
+    line_color = p[0].get_color()
+
+    # plot data slices
+    ng = F_x.n_grow_vect
+    # q_e = -1.602176634e-19   # particle charge (not used)
+    for mfi in F_x:
```

Comment:
We can now, when performance is not the issue, use a cool new global indexing feature to access all boxes, even (MPI-wise) remote ones:
https://pyamrex.readthedocs.io/en/latest/usage/compute.html
<img width="731" height="840" alt="image" src="https://github.com/user-attachments/assets/e1070397-6199-4e6f-a0c1-13ed97e87b16" />

So we can do things now like:
```py
shape = F_x.shape()
F_x_slice_xy = F_x[:,:,shape[2]//2]

plt.imshow(F_x_slice_xy)
plt.show()
```